### DISCUSSION TITLE
routeChangeStart - convert path into a route object

### DISCUSSION THREAD

**Discussion by viinxent** [vercel/next.js#16919] [Help]
Hey guys,

So i'm using `Router.events` to use loading screens on our applications, the problem is per different page we are using skeleton loaders so depending on the page the skeleton loader is different. Is there a way to convert the `path` parameter of `Route.events` i.e `routeChangeStart` into

```
{
  path: '/asfas
  params: {
    param1: 'param1',
    param2: 'param2',
  }
}
```

basically the router object.

I tried getting the `router` props on the `_app` but `on.routeChangeStart` seems like it's not updated and still using the previous route it only changes when it ends `on.routeChangeComplete`

Thanks everyone

**Comment by pitasi** [comment]
Your best bet is to wait for #16524 to be merged. I'm waiting for this feature too :)
